python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -346,6 +327,17 @@ def upstream_key_iterator(self, asset_key: AssetKey) -> Iterator[AssetKey]:
                     queue.append(parent_key)
                     visited.add(parent_key)
 
+    def get_required_neighbors(self, asset_key: AssetKey) -> AbstractSet[AssetKey]:
+        """For a given asset_key, return the set of asset keys that must be materialized at the same time."""
+        if self.required_neighbor_sets is None:
+            raise DagsterInvariantViolationError(
+                "Required neighbor information not set when creating this AssetGraph"
+            )
+        for required_assets in self.required_neighbor_sets:
```

Comment:
If you had a bunch of multi-assets, this would be O(# assets), right? Thoughts on using a dict?

Reply:
mm yeah -- originally I was thinking we'd need to create some unique identifier for the keys of the dictionary, then some mapping from asset key to that unique identifier, but now I realize that's not necessary at all. will update.